File: leetcode/problems/language-specific/go/520.detect-capital.go
/*
 * @lc app=leetcode id=520 lang=golang
 *
 * [520] Detect Capital
 */

// @lc code=start
func detectCapitalUse(word string) bool {
	var isUpperCase = regexp.MustCompile(`^[A-Z]+$`).MatchString
	var isLowerCase = regexp.MustCompile(`^[a-z]+$`).MatchString
	var isTitleCase = regexp.MustCompile(`^[A-Z]{1}[a-z]+$`).MatchString

	return isUpperCase(word) || isLowerCase(word) || isTitleCase(word)
}

// @lc code=end

File: leetcode/problems/problem-specific/520.detect-capital/520.detect-capital.py
#
# @lc app=leetcode id=520 lang=python3
#
# [520] Detect Capital
#

# @lc code=start

# * Counting Solution | O(n) Time | O(1) Space
# * n -> The length of word string


class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        return word.isupper() or word.islower() or word.istitle()


# @lc code=end
